src/utils/ReactReducer.ts

```diff
@@ -0,0 +1,26 @@
+const CHANGE_STATE = 'CHANGE_STATE';
+
+type stateType = { type: string; key?: string; payload: any; }
+
+const changeAction = (payload: any, key: any) => ({
+    type: CHANGE_STATE,
+    key,
+    payload,
+});
+
+export function stateReducer(state: any, {type, payload, key}: stateType) {
+    switch (type) {
+        case CHANGE_STATE:
+            const newState = key ? {[key]: payload} : payload;
+            return {
+                ...state,
+                ...newState,
+            };
+        default:
+            return state;
+    }
+}
+
```

Comment:
We will use **Redux** with our own division into modules. Are you sure you want to continue with the **pseudo-redux**?